internal/constellation/applyinit.go

```diff
@@ -0,0 +1,265 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package constellation
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/edgelesssys/constellation/v2/bootstrapper/initproto"
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	"github.com/edgelesssys/constellation/v2/internal/grpc/grpclog"
+	grpcRetry "github.com/edgelesssys/constellation/v2/internal/grpc/retry"
+	"github.com/edgelesssys/constellation/v2/internal/kms/uri"
+	"github.com/edgelesssys/constellation/v2/internal/retry"
+	"github.com/edgelesssys/constellation/v2/internal/state"
+	"github.com/edgelesssys/constellation/v2/internal/versions"
+	"google.golang.org/grpc"
+)
+
+// InitPayload contains the configurable data for the init RPC.
+type InitPayload struct {
+	MasterSecret    uri.MasterSecret
+	MeasurementSalt []byte
+	K8sVersion      versions.ValidK8sVersion
+	ConformanceMode bool
+}
+
+// GrpcDialer dials a gRPC server.
+type GrpcDialer interface {
+	Dial(ctx context.Context, target string) (*grpc.ClientConn, error)
+}
+
+// Init performs the init RPC.
+func (a *Applier) Init(
+	ctx context.Context,
+	dialer GrpcDialer,
+	state *state.State,
+	payload InitPayload,
+) (
+	*initproto.InitSuccessResponse,
+	[]byte,
+	error,
+) {
```

Comment:
Instead of returning the logs as `[]byte`, we could pass an `io.Writer` to `Init`.
This would allow us to pass a logging interface for the Terraform provider (we probably want to collect the cluster logs in case of failure and log them), or a buffer which we later write to file for the CLI.

This is mostly an implementation detail tho. There shouldn't be much difference in the behavior between the current implementation to if you change it up.